packages/mendel-pipeline/src/cli.js

```diff
@@ -20,6 +20,30 @@ program
     .option('-w, --watch', 'Watch mode', false)
     .parse(process.argv);
 
-// Example usage
-const mendelConfig = config(Object.assign(program, {cwd: path.resolve(process.cwd(), program.args[0])}));
-mendelPipeline(mendelConfig);
+
+const cliOptions = Object.assign(program, {
+    cwd: path.resolve(process.cwd(), program.args[0]),
+});
+
+if (program.watch) {
+    const daemon = new MendelPipelineDaemon(cliOptions);
+    daemon.watch();
+} else {
+    /* TODO:
+        request.get(defaultDeamonAddres + '/status', (error, response) => {
+            if (error && isConnectionRefused(error)) {
+                const daemon = new MendelPipelineDaemon(cliOptions);
+                daemon.run();
+            } else {
+                request({
+                    url: defaultDeamonAddres + '/run',
+                    query: cliOptions,
+                });
+            }
+        });
```

Comment:
Both the Middleware and the UnitTest can use the same API, some endpoints would be for environments + bundles (and understand the bundle config) other APIs would be per environment + entryId and only need to understand the registry.